BreakingBad/CharacterListScreen.swift

```diff
@@ -56,46 +56,56 @@ class CharacterListScreen: UIViewController {
     
      
     private func createCharacterList(from characterList : [CharacterJson]) -> [Character] {
-
         var tempCharacters: [Character] = []
-
         for character in characterList {
-            //let imageURL = URL(string: character.img)
+            let image = defaultImage!
             let birthday: String = character.birthday
             let age = addAgeToBirthday(from: birthday)
-            let image = defaultImage! //getImage(from: imageURL)
             let newCharacter = Character(name: character.name, birthday: age, image: image, nickname: character.nickname, occupations: character.occupation, portrayed: character.portrayed)
             tempCharacters.append(newCharacter)
-            
+            if let imageURL = URL(string: character.img) {
+                getImageAsync(from: newCharacter, imageUrl: imageURL)
+            }
         }
         return tempCharacters
     }
     
     private func addAgeToBirthday(from birthday: String) -> String {
-        if birthday != "Unknown" {
-            
-            let dateFormatter = DateFormatter()
-            dateFormatter.dateFormat = "MM'-'dd'-'YYY'"
-            let date = dateFormatter.date(from: birthday)
-            let now = Date()
-            guard let formattedBirthday: Date = date else {
+            if birthday != "Unknown" {
+                
+                let dateFormatter = DateFormatter()
+                dateFormatter.dateFormat = "MM'-'dd'-'YYY'"
+                let date = dateFormatter.date(from: birthday)
+                let now = Date()
+                guard let formattedBirthday: Date = date else {
+                    return birthday
+                }
+                let calendar = Calendar.current
+                let ageComponents = calendar.dateComponents([.year], from: formattedBirthday, to: now)
+                let age = birthday + " (" + String(ageComponents.year!) + ")"
+                return age
+            }
+            else {
                 return birthday
             }
-            let calendar = Calendar.current
-            let ageComponents = calendar.dateComponents([.year], from: formattedBirthday, to: now)
-            let age = birthday + " (" + String(ageComponents.year!) + ")"
-            return age
         }
-        else {
-            return birthday
+    
+    private func getImageAsync(from character: Character, imageUrl: URL) {
+        DispatchQueue.global(qos: .userInitiated).async {
+            let characterImage = self.getImageData(imageUrl)
+            character.image = characterImage!
```

Comment:
Force unwrap code smell! :) 

What you are saying here is the optional `characterImage` will NEVER be nil. But is that true? No. It's possible the getImageData method may fail for some reason, returning nil, and that would crash your app. 

Instead of force unwrapping you want to a) unwrap and then assign the property, and if the unwrap fails - ie if its nil, then you want to assign the `defaultImage` as the image.